cabal-install/Distribution/Client/Sandbox.hs

```diff
@@ -453,12 +453,19 @@ sandboxDeleteSource verbosity buildTreeRefs _sandboxFlags globalFlags = do
   indexFile            <- tryGetIndexFilePath (pkgEnvSavedConfig pkgEnv)
 
   withRemoveTimestamps sandboxDir $ do
-    Index.removeBuildTreeRefs verbosity indexFile buildTreeRefs
+    removedRefs <- Index.removeBuildTreeRefs verbosity indexFile buildTreeRefs
+    when (length buildTreeRefs > length removedRefs) $
+      die $ "Skipping nonregistered sources: " ++ (showL $ prune buildTreeRefs removedRefs)
```

Comment:
I think this will cause corruption of the timestamps file when some refs were removed and some were not.